<?xml version="1.0"?>
<launch>

<arg name="args" default=""/>
  
  <include file="$(find larics_gazebo_worlds)/launch/start.launch">
    <arg name="world" value="$(find larics_gazebo_worlds)/worlds/floor_plan_simple.world"/>
    <arg name="args" default="$(arg args)"/>
</include>

<!-- these are the arguments you can pass this launch file, for example paused:=true -->
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="false"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>

  <arg name="enable_logging" default="true"/>
  <arg name="enable_ground_truth" default="true"/>
  <arg name="log_file" default="vpc_mmcuav"/>

  <arg name="model_type" default="mmcuav" />

  <include file="$(find mmuav_description)/launch/spawn_vpc_mmcuav.launch">
    <arg name="model_type" value="$(arg model_type)" />
  </include>
  
   <!-- Start control -->
  <include file="$(find mmuav_control)/launch/vpc_mmcuav_control.launch"/>

  <!-- Start attitude height control -->
<include file="$(find mmuav_control)/launch/vpc_mmcuav_attitude_position_control.launch"/>

	<node name="Calculating_pomdp_server" pkg="pomdp_paper_action" type="Calculating_pomdp_server.py" output="screen"/>
	<node name="beacon" pkg="pomdp_paper_action" type="beacon.py" output="screen"/>
	<node name="goTo" pkg="pomdp_paper_action" type="goToServer.py" output="screen"/>
	<node name="listen" pkg="pomdp_paper_action" type="ListenServer.py" output="screen"/>
	<node name="Master_client" pkg="pomdp_paper_action" type="MasterClient.py" output="screen"/>

</launch>
